File: Easy/ToLower Case.py
def toLowerCase(s: str) -> str:
    # Initialize an empty string to store the converted characters
    lowercase = ""

    # Loop through each character in the input string
    for char in s:
        # Check if the character is uppercase
        if 'A' <= char <= 'Z':
            # Convert the uppercase character to lowercase using ASCII manipulation
            lowercase += chr(ord(char) + 32)
        else:
            # If the character is not uppercase, keep it unchanged
            lowercase += char 

    # Return the converted string with lowercase characters
    return lowercase

# Test the function
string = "UpperCase"
print(toLowerCase(string))

File: Easy/ToLower Case.cpp
#include <iostream>

// Function to convert a string to lowercase
// Returns the lowercase version of the input string

std::string toLowerCase(const std::string& s) {
    // Initialize an empty string to store the lowercase result
    std::string result = "";

    // Iterate through each character in the input string
    for (char i : s) {
        // Check if the character is an uppercase letter
        if (i >= 'A' && i <= 'Z') {
            // Convert uppercase to lowercase using ASCII values and append to the result
            result += static_cast<char>(i + 32);
        } else {
            // Keep non-uppercase characters unchanged and append to the result
            result += i;
        }
    }

    // Return the final lowercase result
    return result;
}

int main() {
    // Read the string input from the user
    std::string inputString;
    std::getline(std::cin, inputString);

    // Convert the input string to lowercase by calling the toLowerCase function
    std::string lowercaseString = toLowerCase(inputString);

    // Display the lowercase string
    std::cout << lowercaseString << std::endl;

    return 0;
}
